sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Statistics.scala

```diff
@@ -184,6 +196,26 @@ object ColumnStat extends Logging {
   }
 
   /**
+   * Converts from string representation of external data type to the corresponding Catalyst data
+   * type.
+   */
+  private def fromExternalString(s: String, name: String, dataType: DataType): Any = {
+    dataType match {
+      case BooleanType => s.toBoolean
+      case _: IntegralType => s.toLong
+      case DateType => DateTimeUtils.fromJavaDate(java.sql.Date.valueOf(s)).toLong
```

Comment:
hm.. I think it's better to keep min/max value in ColumnStat just the same as internal type, i.e. don't cast short/int to long. This causes less confusion. Besides, we'll use decimal to unify their comparison and computation logics anyway.